docs/src/prototypes/chatMessageWithPopover/Popover.tsx

```diff
@@ -0,0 +1,118 @@
+
+export interface PopoverProps {
+  className?: string
+}
+
+class Popover extends React.Component<PopoverProps> {
+  render() {
+    return (
+      <Menu
+        styles={{
+          transition: 'opacity 0.2s',
+          position: 'absolute',
+          top: '-20px',
+          right: '5px',
+          background: '#fff',
+          boxShadow: '0px 2px 4px #ddd',
+          borderRadius: '.3rem',
+          opacity: 0,
+
+          '& .smile-emoji': {
+            display: 'none',
+          },
+
+          '&.focused .smile-emoji': {
+            display: 'flex',
+          },
+
+          '&:hover .smile-emoji': {
+            display: 'flex',
+          },
+
+          '& a:focus': {
+            textDecoration: 'none',
+            color: 'inherit',
+          },
+
+          '& a': {
+            color: 'inherit',
+          },
+        }}
+        iconOnly
+        className={this.props.className}
+        items={[
+          { key: 'smile', icon: 'smile', className: 'smile-emoji' },
+          { key: 'smile2', icon: 'smile', className: 'smile-emoji' },
+          { key: 'smile3', icon: 'smile', className: 'smile-emoji' },
+          { key: 'a', icon: 'thumbs up' },
+          { key: 'c', icon: 'ellipsis horizontal' },
+        ]}
+        renderItem={renderContextMenu}
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+        accessibility={toolbarBehavior}
+        data-is-focusable={true}
+      />
+    )
+  }
+}
+
+export default Popover
+
+const renderContextMenu = (MenuItem, props) => {
+  if (props.icon !== 'ellipsis horizontal') {
+    return <MenuItem {...props} />
+  }
+
+  return (
+    <Popup
+      key={props.key}
+      position="below"
+      accessibility={popupFocusTrapBehavior}
+      trigger={<MenuItem {...props} />}
+      content={
+        <div
+          style={{
+            background: '#fff',
+            boxShadow: '0 0.2rem 1.6rem 0 rgba(37,36,35,.3)',
+            borderRadius: '.3rem',
+            marginTop: '5px',
+          }}
+        >
+          <Menu
+            vertical
+            pills
+            className="actions"
+            items={[
+              { key: 'bookmark', icon: 'folder', content: 'Save this message' },
+              { key: 'linkify', icon: 'linkify', content: 'Copy link' },
+              { key: 'translate', icon: 'translate', content: 'Translate' },
+            ]}
+          />
+        </div>
+      }
+    />
+  )
+}
+
+export const handleFocus = e => {
+  const currentTarget = e.currentTarget
+  if (!currentTarget.classList.contains('focused')) {
+    currentTarget.classList.add('focused')
+  }
+}
+
+export const handleBlur = e => {
+  const currentTarget = e.currentTarget
+  const relatedTarget = e.relatedTarget
+
+  if (currentTarget.contains(relatedTarget)) {
+    if (!currentTarget.classList.contains('focused')) {
+      currentTarget.classList.add('focused')
+    }
+  } else {
+    currentTarget.classList.remove('focused')
+  }
```

Comment:
We should not manipulate DOM directly, we can introduce a state variable that will define applied props or classes in `render()`

Reply:
We can treat these methods ```handleBlur``` and ```handleFocus``` as additional lib (can move them to separate file), which defines if element got focus or not, by adding an additional class to target. These methods work both equally for message container and popover itself.
+ we avoid re-rendering component on each state change (focus/blur events can be triggered many-many times)

Reply:
+1
why not use `:focus` pseudoselector instead of this `focused` class?

Reply:
because if the focus is on the child element, like popover's menu item, we should still show it and keep this "focused" state.
Anyway, going to move it to state based on @layershifter comment

Reply:
ah I read the code more carefully, got it; too bad IE and Edge don't have support for [`:focus-within`](https://css-tricks.com/almanac/selectors/f/focus-within/)